tools/mo/openvino/tools/mo/ops/BlockLSTM.py

```diff
@@ -28,22 +28,47 @@ def infer(node: Node):
                 2          | b: The bias vector
                 3          | h_prev: Previous/initial hidden state
                 4          | cs_prev: Value of the initial cell state
-         """
-        assert len(node.in_nodes()) == 5
 
+         MO output edges:    |   Description:
+                0           | hs: Output data / output hidden states concatenated over the whole time sequence
+                1           | cs: Output cell states concatenated over the whole time sequence
         """
-        MO output edges:    |   Description:
-                0           | cs: Output data / output hidden states concatenated over the whole time sequence
-                1           | h: Output cell states concatenated over the whole time sequence
-        """
+
+        node_name = node.soft_get('name', node.id)
+        connected_in_ports = [port_idx for port_idx, port in node.in_ports().items() if not port.disconnected()]
+        connected_out_ports = [port_idx for port_idx, port in node.out_ports().items() if not port.disconnected()]
+        assert len(connected_in_ports) >= 5, "Internal Model Optimizer Error or unsupported BlockLSTM node {}. " \
+                                             "MO expects five inputs for BlockLSTM".format(node_name)
+        assert len(connected_out_ports) <= 2, "Internal Model Optimizer Error or unsupported BlockLSTM node {}. " \
+                                              "MO expects at most two outputs for BlockLSTM".format(node_name)
+
+        x_shape = node.in_port(0).data.get_shape()
+        w_shape = node.in_port(1).data.get_shape()
+        b_shape = node.in_port(2).data.get_shape()
+
+        time_len = dynamic_dimension
+        batch_size = dynamic_dimension
+        if len(x_shape) > 2:
+            time_len = x_shape[0]
+            batch_size = x_shape[1]
+
+        hidden_size_output = dynamic_dimension
+        if len(b_shape) > 0 and b_shape[0] is not dynamic_dimension:
+            hidden_size_output = b_shape[0] // 4
+        elif len(w_shape) > 1 and w_shape[1] is not dynamic_dimension:
+            hidden_size_output = w_shape[1] // 4
 
         assert len(node.out_nodes()) in [1, 2]
```

Comment:
don't think this assert is needed since you do check outputs on line 42

Reply:
Agree

Reply:
Done